### PR TITLE
refactor(@desktop/profile-settings-advanced): advanced tab of the profile settings updated

### PR DESCRIPTION
**Pull request by saledjenic**
The first commit introduces `globalUtils` instance (changed old `utilsModel`), second commit refers to the advanced module which is a submodule of the profile section module, third commits adds feature of adding custom networks.

The thing which is missing (but actually that's not a part of the advanced module responsibility) is responding to the changes made in the advanced tab, for which we don't need to restart the app. 
For example: 
- if user enables communities in the advanced tab we should respond to that change and without restarting the app add "Communities" action to the context menu available on right click in the "Chat" section and add all communities a logged in user participates in (not completely sure about the latter, but would be good to have that as well).
- or if user enables "Online users" we should respond to that change and without restarting the app add option for opening users list in chats/channels where that is possible.

Adding in a loop here @iurimatias cause this change affects settings/networks/nodes.

@richard-ramos since you're familiar enough with that part please pay attention to [this part](https://github.com/status-im/status-desktop/blob/ca8bd034a51fa089e605a74de979fd91cf204c15/src/app_service/service/node_configuration/service.nim#L145) in the old code base we didn't take into consideration what waku version we are setting bloom filter for, I added that now, corresponding old code is [here](https://github.com/status-im/status-lib/blob/7095b80cb6204c7b88012edbb3d97f59df330ddc/status/statusgo_backend/settings.nim#L196), as you may see we always set it in the past as it's always V1.

### CODE REVIEW

**Review by jrainville — APPROVED** [review]
Nice job

File: src/app_service/service/node_configuration/service.nim [hunk 1]
```diff
@@ -46,20 +46,45 @@ method init*(self: Service) =
...
+    return BLOOM_LEVEL_NORMAL
+
+  if (bloomFilterMode):
+    if(fullNode):
+      return BLOOM_LEVEL_FULL
+    else:
+      return BLOOM_LEVEL_NORMAL
+  else:
+    return BLOOM_LEVEL_LIGHT
+
```
Comment (richard-ramos) [inline src/app_service/service/node_configuration/service.nim:86]: Bloom levels only apply to WakuV1. This function should either return empty or throw an exception if called when the current waku version is 2

File: src/app_service/service/node_configuration/service.nim [hunk 2]
```diff
@@ -73,45 +98,65 @@ method setWakuVersion*(self: Service, wakuVersion: int) =
...
+    newConfiguration.WakuConfig.BloomFilterMode = bloomFilterMode
+    newConfiguration.WakuConfig.FullNode = fullNode
+    newConfiguration.WakuConfig.LightClient = not fullNode
+  elif wakuVersion == WAKU_VERSION_2:
+    newConfiguration.WakuV2Config.BloomFilterMode = bloomFilterMode
+    newConfiguration.WakuV2Config.FullNode = fullNode
+    newConfiguration.WakuV2Config.LightClient = not fullNode
+  
+  return self.saveConfiguration(newConfiguration)
+
```
Comment (richard-ramos) [inline src/app_service/service/node_configuration/service.nim:158]: BloomLevel only applies to wakuV1, the condition for WAKU_VERSION_2 is not necessary.
Perhaps an exception must be thrown if the function is executed when waku2 is enabled?

**Comment by saledjenic** [comment]
@richard-ramos thank you for a reply, that's why I asked you, I removed parts for reading/setting bloom filter in case of WakuV2, should be all good now.

**Review by richard-ramos — APPROVED** [review]

**Pull request merged**